src/current/v23.1/connection-pooling.md

```diff
@@ -25,37 +25,149 @@ Each time an application reads or writes data, it will request one of the connec
+For multi-region clusters, create a connection pool per region, and size the maximum connection pool for each region in your cluster using the same formula as a single-region cluster.
 
-In addition to setting a maximum connection pool size, set the maximum number of idle connections if possible. Cockroach Labs recommends setting the maximum number of idle connections to the maximum pool size. While this uses more memory, it allows many connections when concurrency is high without having to create a new connection for every new operation.
+For example, if you have 3 regions in your cluster, and each region has 12 vCPUs, create a connection pool for each region, with each connection pool having a maximum pool size of 48 (12 [processor cores] * 4).
 
 {% include {{page.version.version}}/sql/server-side-connection-limit.md %} This may be useful in addition to your connection pool settings.
 
-## Validating connections in a pool
+</section>
 
-After a connection pool initializes connections to CockroachDB clusters, those connections can occasionally break. This could be due to changes in the cluster topography, or rolling upgrades and restarts, or network disruptions. CockroachDB {{ site.data.products.cloud }} clusters periodically are restarted for patch version updates, for example, so previously established connections would be invalid after the restart.
+Cockroach Labs strongly recommends that your application instance run in the same region as your CockroachDB cluster.
 
-Validating connections is typically handled automatically by the connection pool. For example, in HikariCP the connection is validated whenever you request a connection from the pool, and the `keepaliveTime` property allows you to configure an interval to periodically check if the connections in the pool are valid. Whatever connection pool you use, make sure connection validation is enabled when running your application.
+## Monitor SQL connections
+
+To validate that your connection pool is correctly configured, monitor the SQL connection metrics and graphs on your cluster.
+
+### Monitor new connections
+
+<section class="filter-content" markdown="1" data-scope="dedicated selfhosted">
+The [`sql.new_conns` metric]({% link {{ page.version.version }}/metrics.md %}#available-metrics) and [SQL Connection Rate graph]({% link {{ page.version.version }}/ui-sql-dashboard.md %}#sql-connection-rate) expose the number of new SQL connections per second.
+</section>
+
+<section class="filter-content" markdown="1" data-scope="serverless">
+The [SQL Connection Attempts graph]({% link cockroachcloud/metrics-page.md %}#identify-sql-problems) shows the number of new SQL connections per second.
+</section>
+
+A misconfigured connection pool will result in a very large number of new connections per second.
```

Comment:
If I'm a customer, I'd immediately ask what is "a very large number". 10/second? 100/second?

This advice should be the same for Serverless/Dedicated/self-hosted. I suggested a 100:1 ratio as a basic rule of thumb. We can use something different, but I think we should be consistent across the products. IMO, @dikshant and team should own the decision about what to recommend here.

Reply:
maybe we can say something like "A misconfigured connection pool will result in most database operations requiring a new connection to be established."

Reply:
@andy-kimball is there a reason 100 is a good rule of thumb? Most drivers and ORMs that set a max number of connections default to a fairly small number of connections (2-10). There are exceptions like npgsql which go as high as 100 but the connection lifetime tends to be long enough (30 mins). Which makes it pretty difficult to hit 100 connections/sec. Plus I have seen some customers who run their applications with 0.5 vCPUs, which will definitely choke when trying to create too many connections at once (around 10).

We can make the consequences a bit clear for now:
> A misconfigured connection pool will result in most database operations requiring a new connection to be established which will increase query latency.

I'll do some investigating as part of https://github.com/cockroachdb/docs/issues/17820 and add a more concrete number there.